test/dialects/spark3_grammar_test.py

```diff
@@ -0,0 +1,97 @@
+"""Test some specific spark3 grammars."""
+import logging
+from typing import List, Callable, Any
+
+import pytest
+
+from sqlfluff.core import FluffConfig, dialect_selector
+from sqlfluff.core.dialects import Dialect
+from sqlfluff.core.parser import Lexer, RawSegment
+from sqlfluff.core.parser.context import RootParseContext
+
+
+@pytest.fixture(scope="function")
+def fresh_spark3_dialect() -> Dialect:
+    """Expand the ansi dialect for use."""
+    return dialect_selector("spark3")
+
+
+def lex_segments(fragment: str):
+    """For given test examples, check successful parsing."""
+    config = FluffConfig(overrides=dict(dialect="spark3"))
+    tokens, lex_vs = Lexer(config=config).lex(fragment)
+    # From just the initial parse, check we're all there
+    assert "".join(token.raw for token in tokens) == fragment
+    # Check we don't have lexing issues
+    assert not lex_vs
+    return tokens
+
+
+@pytest.mark.parametrize(
+    ["segment_class", "match_or_parse", "fragment"],
+    [
+        (t[1], b, t[0])
+        for t in [
+            ("as t ( col1 , col2 )", "AliasExpressionSegment"),
+            ("as t \t ( col1 , col2 )", "AliasExpressionSegment"),
+            ("from t", "FromClauseSegment"),
+            ("from t as u", "FromClauseSegment"),
+            ("from values ( 1 , 2 )", "FromClauseSegment"),
+            ("from values 1", "FromClauseSegment"),
+            ("from values 1 , 2 ", "FromClauseSegment"),
```

Comment:
this example is what ultimately allowed me to realize what why it wouldn't parse; i was getting very confused because `values 1,2` was correctly parsing as a `ValuesClauseSegment` and a `FromExpressionSegment`